### PR TITLE
aq.py: fix aqhost option ignored

### LINKED ISSUE #96: aq.py ignores aqhost in config file outside MS
`aq.py` sets unconditionnally the default `aqhost` to the current host outside MS, instead of taking into account the defaults defined in aq conf file.

### PR DESCRIPTION
**Pull request by jouvin**
Fixes #96

Change-Id: Ib70cde2d494f2afe5eaa6546d84a8e49f4107a75

### CODE REVIEW

File: bin/aq.py [hunk 1]
```diff
@@ -347,8 +347,8 @@ def get_default_opts(auth_option, conf_file=None):
         default_aqhost = defaultOpts.get('aqhost') or 'nyaqd1'
         default_aqservice = 'cdb'
     else:
-        default_aqhost = socket.gethostname()
-        default_aqservice = get_username()
+        default_aqhost =  defaultOpts.get('aqhost') or socket.gethostname()
+        default_aqservice =  defaultOpts.get('aqservice') or get_username()
 
     host = globalOptions.get('aqhost') or os.environ.get('AQHOST', None) or \
```
Comment (urbonegi) [inline bin/aq.py]: Actually aqservice is already not ignoring aq.conf default aqservice value:
https://github.com/jouvin/aquilon/blob/3a416e34347731196868be30156866828f78de33/bin/aq.py#L377

I suggest to leave this line alone and just add additional 'or defaultOpts.get('aqhost')'  after globalOptions.get('aqhost') in the line #353 for host.

Comment (jouvin) [reply bin/aq.py]: @urbonegi I missed the fact that `defaultOpts.get('aqservice')` was called later. I have the feeling that `default_aqservice` shoud not be defined in the `else` clause that I updated as it is a source of confusion. I'll update the PR.

File: bin/aq.py [hunk 2]
```diff
@@ -347,7 +347,7 @@ def get_default_opts(auth_option, conf_file=None):
         default_aqhost = defaultOpts.get('aqhost') or 'nyaqd1'
         default_aqservice = 'cdb'
     else:
-        default_aqhost = socket.gethostname()
+        default_aqhost =  defaultOpts.get('aqhost') or socket.gethostname()
```
Comment (urbonegi) [inline bin/aq.py]: please leave the 'default_aqhost' as it is.. and just add the 'defaultOpts.get('aqhost')' to the line #353  after global options. Same as it is done for service:
host = globalOptions.get('aqhost') or defaultOpts.get('aqhost') or os.environ.get('AQHOST', None) or \
        default_aqhost

**Comment by urbonegi** [comment]
Hey @jouvin just following up on this one. Could you please add the fix as per my comment above to be consistent in the behavior. We then will include this into next release :)

**Comment by jouvin** [comment]
@urbonegi sorry for the delay... I did what you recommended! BTW, reading again the code, I'm a bit surprise by the evaluation order of host, port, service possible definitions. I'd expect the environment variable to act as an override value rather than a default one, evaluated after the keys found in the configuration file. Am I wrong?

File: bin/aq.py [hunk 3]
```diff
@@ -347,11 +347,11 @@ def get_default_opts(auth_option, conf_file=None):
         default_aqhost = defaultOpts.get('aqhost') or 'nyaqd1'
         default_aqservice = 'cdb'
     else:
-        default_aqhost = socket.gethostname()
+        default_aqhost =  socket.gethostname()
```
Comment (urbonegi) [inline bin/aq.py]: unnecessary space :) please remove.

Comment (urbonegi) [inline bin/aq.py]: unnecessary space :) please remove.

Comment (urbonegi) [inline bin/aq.py]: unnecessary space :) please remove.

Comment (urbonegi) [inline bin/aq.py]: unnecessary space :) please remove.

File: bin/aq.py [hunk 4]
```diff
@@ -347,11 +347,11 @@ def get_default_opts(auth_option, conf_file=None):
...
         default_aqservice = get_username()
 
-    host = globalOptions.get('aqhost') or os.environ.get('AQHOST', None) or \
-        default_aqhost
+    host = globalOptions.get('aqhost') or defaultOpts.get('aqhost') or \
+           os.environ.get('AQHOST', None) or default_aqhost
```
Comment (urbonegi) [inline bin/aq.py]: Yes you are right - env variables should override the default. Therefore they should go first :) I do not remember if i told u an order - but it should be:
```
host = globalOptions.get('aqhost') or os.environ.get('AQHOST', None) or \
 defaultOpts.get('aqhost') or default_aqhost
```

**Comment by urbonegi** [comment]
Ok looks good now - just pne thing though - please add more verbose message of the commit that is introduced to have better proper git history :) also the link to the github issue should be more verbose. for example Addresses Issue: Github #96

**Comment by jouvin** [comment]
@urbonegi is it matching your expectation now? Note that the 'Fixes #96' is "idiomatic" and is required for the issue to be closed when the PR is merged.

**Comment by xaf** [comment]
Hi @jouvin 
Your commit message seems to be a bit blurry about what the commit really does, and requires one to look at the code to understand what's happening.

I would suggest the following:
```
aq.py: add check in default options for 'host' value

Other options such as 'aqservice' and 'port' are set by checking,
among others, the 'defaultOpts' dict, but for 'host' we were only
checking the 'globalOpts' dict. This fixes that.
This also removes the mention to 'get_username()' to set 'aqservice'
as it is already done in the if-else block setting the 'default_aqservice'.

Fixes #96
```

Moreover, could you please ensure the alignment in line 354 of your patch is done properly ? You currently aligned with the previous line, which gives:

```
   host = globalOptions.get('aqhost') or os.environ.get('AQHOST', None) or \
          defaultOpts.get('aqhost') or default_aqhost
```

Instead of the way it should be: (next line + 1 indent)

```
   host = globalOptions.get('aqhost') or os.environ.get('AQHOST', None) or \
       defaultOpts.get('aqhost') or default_aqhost
```

It would also be fine to do:

```
   host = (globalOptions.get('aqhost') or os.environ.get('AQHOST', None) or
           defaultOpts.get('aqhost') or default_aqhost)
```

Thanks! :)

**Comment by jouvin** [comment]
Done! Hopefully it is good now!

**Comment by xaf** [comment]
Thank you! This should be picked up soon :)

**Comment by urbonegi** [comment]
Merged https://github.com/quattor/aquilon/commit/8d5ca8143b2ac85a6a9002c6f3fdf8cf00b909ce

**Pull request closed without merging**